src/textual/widgets/_data_table.py

```diff
@@ -1215,6 +1225,22 @@ def _update_dimensions(self, new_rows: Iterable[RowKey]) -> None:
                 content_width = measure(console, renderable, 1)
                 column.content_width = max(column.content_width, content_width)
 
+            if row.auto_height:
+                auto_height_rows.append((row, cells_in_row))
+
+        for row, cells_in_row in auto_height_rows:
```

Comment:
Since we will need to render the cells anyway, can we get the height without rendering and throwing away the result ?

Reply:
I.e., can we store the renderings?

Reply:
I'm not sure how feasible it is.
I currently render the cell in a quick & dirty way just to count lines.

I looked into modifying `DataTable._render_cell`, which is responsible for rendering a cell and caching its result.
However, `_render_cell` needs a lot of information that I don't have computed, of which the base style is the more troublesome to compute.
`DataTable._render_line_in_row` spends ~90 lines on that, and I am not sure whether it's faster to do this simpler throwaway rendering here or if we should go over all that trouble at this point.

Reply:
I think its a clear win, and worth perservering.

Reply:
See c9ae50d.